src/ucp/wireup/wireup_cm.c

```diff
@@ -741,10 +767,33 @@ static void ucp_cm_disconnect_cb(uct_ep_h uct_cm_ep, void *arg)
 {
     ucp_ep_h ucp_ep            = arg;
     uct_worker_cb_id_t prog_id = UCS_CALLBACKQ_ID_NULL;
+    ucp_worker_h worker        = ucp_ep->worker;
+    uct_ep_h uct_ep;
+    int discard_uct_ep;
 
     ucs_trace("ep %p: CM remote disconnect callback invoked, flags 0x%x",
               ucp_ep, ucp_ep->flags);
 
+    uct_ep = ucp_ep_get_cm_uct_ep(ucp_ep);
+    if (uct_ep == NULL) {
+        UCS_ASYNC_BLOCK(&worker->async);
+        discard_uct_ep = ucp_worker_discard_uct_ep_in_progress(worker,
+                                                               uct_cm_ep);
+        UCS_ASYNC_UNBLOCK(&worker->async);
+
+        if (discard_uct_ep) {
+            ucs_diag("ep %p: UCT EP %p for CM lane doesn't exist, it"
+                     " has already been discarded", ucp_ep, uct_cm_ep);
+            return;
+        }
+
+        ucs_fatal("ep %p: UCT EP for CM lane doesn't exist", ucp_ep);
```

Comment:
i think `ucp_worker_discard_uct_ep_in_progress` for CM lane does not make sense since flush of CM lane returns OK immediately => the UCT_EP is destroyed tool
otherwise there is low chance to catch this fatal.

Reply:
I catched this with TCP
it could be possible, since we do flush and destroy from progress fucntions

Reply:
ic